app/src/main/java/in/co/rajkumaar/amritarepo/aums/activities/CourseResourcesActivity.java

```diff
@@ -0,0 +1,496 @@
+                                public void run() {
+                                    Toast.makeText(CourseResourcesActivity.this, "No resources uploaded in this folder!", Toast.LENGTH_LONG).show();
+                                    finish();
+                                }
+                            });
+                        }
+                    } else {
+                        String lastRes = tRows.last().select("td:nth-child(1) > a").first().text().trim();
+                        for (Element row : tRows) {
+                            String resName = row.select("td:nth-child(1) > a").first().text().trim();
+                            String resUrl = row.select("td:nth-child(1) > a").first().attr("href").trim();
+                            String resType;
+                            String oneUp = "../";
+                            if (!(resUrl.equals(oneUp))) {
+                                if (resUrl.substring(resUrl.length() - 1).equals("/")) {
+                                    resType = "Folder";
+                                } else {
+                                    resType = resUrl.substring(resUrl.lastIndexOf('.') + 1);
+                                }
+                                resUrl = folder + resUrl;
+                                courseResourceList.add(new CourseResource(resName, resUrl, resType));
+                                if (resName.equals(lastRes)) {
+                                    courseResourceStack.add(courseResourceList);
+                                    CourseResAdapter courseResAdapter = new CourseResAdapter(CourseResourcesActivity.this, courseResourceList);
+                                    list.setAdapter(courseResAdapter);
+                                    if (firstEntry) {
+                                        findViewById(R.id.progressBar).setVisibility(GONE);
+                                        list.setVisibility(View.VISIBLE);
+                                        firstEntry = false;
+                                    } else {
+                                        progressDialog.dismiss();
+                                    }
+                                }
+                            }
+                        }
+                    }
+                } catch (Exception e) {
+                    Toast.makeText(CourseResourcesActivity.this, getString(R.string.site_change), Toast.LENGTH_LONG).show();
+                    finish();
+                }
+            }
+
+            @Override
+            public void onFailure(int statusCode, Header[] headers, byte[] responseBody, Throwable error) {
+                Utils.showUnexpectedError(CourseResourcesActivity.this);
+                finish();
+            }
+        });
+    }
+
+    @Override
+    public void onBackPressed() {
+        if (courseResourceStack.size() == 1) {
+            finish();
+        } else {
+            int index = courseResourceStack.size() - 1;
```

Comment:
Better use a `Stack<ArrayList<CourseResource>>` so you don't have to calculate the index and then remove. Just take the top() and do stuffs, and do a pop().
I think you can do the same some `curFolder` too.